plugins/modules/identity/ipa/ipa_config.py

```diff
@@ -181,6 +186,13 @@
     ipa_host: localhost
     ipa_user: admin
     ipa_pass: supersecret
+
+- name: Ensure the SELinux user map order is set
+  community.general.ipa_config:
+    ipaselinuxusermaporder: "guest_u:s0$xguest_u:s0$user_u:s0$staff_u:s0-s0:c0.c1023$unconfined_u:s0-s0:c0.c1023"
```

Comment:
Out of curiosity: does it make sense to convert this option to a list of string (or even dict), so this can be specified in a more readable form?

Reply:
I concur.  It will also need a note in the documents about whether the first element in the list is the highest or lowest priority. 
 I would recommend the first being the lowest to keep in line with the API browser for IPA ("Order in increasing priority of SELinux users").

Reply:
I am not sure about what format to use then. What format are you thinking of @felixfontein ?

Something like the following ('list of strings' instead of a 'string') and then have the module put it together again to be `$` separated [1]?:

```yml
ipaselinuxusermaporder:
- "guest_u:s0"
- "xguest_u:s0"
- "user_u:s0"
- "staff_u:s0-s0:c0.c1023"
- "unconfined_u:s0-s0:c0.c1023"
```

[1] On [this page](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/linux_domain_identity_authentication_and_policy_guide/config-selinux) I found that the values are `$` separated: ` The individual user entries are separated with a dollar sign ($).`

Reply:
> I am not sure about what format to use then. What format are you thinking of @felixfontein ?
> 
> Something like the following ('list of strings' instead of a 'string') and then have the module put it together again to be `$` separated [1]?:
> 
> ```yaml
> ipaselinuxusermaporder:
> - "guest_u:s0"
> - "xguest_u:s0"
> - "user_u:s0"
> - "staff_u:s0-s0:c0.c1023"
> - "unconfined_u:s0-s0:c0.c1023"
> ```
> 
> [1] On [this page](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/linux_domain_identity_authentication_and_policy_guide/config-selinux) I found that the values are `$` separated: ` The individual user entries are separated with a dollar sign ($).`

Yes, if he is thinking similar to me on the topic, a list of strings is what he would suggest.  The module should take care of the 'lower level' IPA details like separating them with `$` or whatever else is needed to accomplish the mapping order setting.  Since the order matters, you would need to define to users the precedence order.

Reply:
A list of dict could mean something like
```
- user: guest_u
   mls: s0
- user: xguest_u
   mls: s0
- user: staf_u
   msl: s0-c0
   mcs: 0.c1023
```
I'm not sure how useful this would be, though. (I'm not configuring IPA :) )

But allowing a list of strings (and joining them together by `$` in the module) is definitely a good idea IMO.

Reply:
I just updated the PR to allow a list of strings. As a user IMO it's not that useful to have the additional granularity for specifying the user, mls and mcs as separate parts.